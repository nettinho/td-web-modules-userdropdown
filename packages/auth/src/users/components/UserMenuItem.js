import React from "react";
import PropTypes from "prop-types";
import { Link, withRouter } from "react-router-dom";
import { Icon, Menu } from "semantic-ui-react";
import { bindActionCreators, compose } from "redux";
import { connect } from "react-redux";
import { FormattedMessage } from "react-intl";
import { hideSidebar } from "@truedat/core/routines";
import { ActiveRoute } from "@truedat/core/components/ActiveRoute";
import routes from "../routes";
import groupRoutes from "../../groups/routes";

export const UserMenuItem = ({ location }) => (
  <Menu.Item
    as={Link}
    to={routes.USER_LIST}
    name="users"
    active={
      ActiveRoute(location, routes.USER_LIST) ||
      ActiveRoute(location, groupRoutes.GROUP_LIST)
    }
  >
    <Icon name="users" />
    <FormattedMessage id="navigation.admin.users" />
  </Menu.Item>
);

UserMenuItem.propTypes = {
  location: PropTypes.object
};

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ hideSidebar }, dispatch)
});

export default compose(
  withRouter,
  connect(
    null,
    mapDispatchToProps
  )
)(UserMenuItem);
