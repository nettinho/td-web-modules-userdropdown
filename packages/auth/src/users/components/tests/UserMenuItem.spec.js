import React from "react";
import { shallowWithIntl } from "intl-stub";
import { UserMenuItem } from "../UserMenuItem";

describe("<UserMenuItem />", () => {
  const hideSidebar = jest.fn();
  const location = { pathname: "/" };
  const props = { hideSidebar, location };

  it("matches the latest snapshot", () => {
    const wrapper = shallowWithIntl(<UserMenuItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
