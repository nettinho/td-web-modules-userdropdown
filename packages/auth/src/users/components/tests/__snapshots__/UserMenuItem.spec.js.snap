// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserMenuItem /> matches the latest snapshot 1`] = `
<MenuItem
  active={false}
  as={[Function]}
  name="users"
  to="/users"
>
  <Icon
    as="i"
    name="users"
  />
  <FormattedMessage
    id="navigation.admin.users"
    values={Object {}}
  />
</MenuItem>
`;
